cpp/src/stream_compaction/stable_distinct.cu

```diff
@@ -58,13 +66,22 @@ std::unique_ptr<table> stable_distinct(table_view const& input,
     return markers;
   }();
 
-  return cudf::detail::copy_if(
-    input,
-    [output_markers = output_markers.begin()] __device__(auto const idx) {
-      return *(output_markers + idx);
-    },
-    stream,
-    mr);
+  return cudf::detail::apply_boolean_mask(
```

Comment:
Using `apply_boolean_mask` incurs no noticeable performance penalty, but speeds up compile time. We already have to compile kernels for `apply_boolean_mask`, so it's good to reuse them rather than add more Thrust kernels.

Reply:
This was inspired by the RAFT talk on reducing compile times. I think there are a ton of places like this in cudf that could shrink our kernel footprint.

Reply:
I vaguely remember that we have a lot of places that have similar compiling burdensome not just with `copy_if` but also `detail::gather`. I believe that we can easily convert calls to `detail::gather` into using `apply_boolean_mask` too.

Reply:
BTW, the name `apply_boolean_mask` is a bit misleading to me. What is the outcome of such "applying"? We are copying out of the input, not applying anything to it (in place). So maybe it should have a better name. Renaming it into another overload of `copy_if` could be an option.

Reply:
I think that could be sensible but is out of scope for this PR.

Reply:
Filed an issue: https://github.com/rapidsai/cudf/issues/13405